types/three/src/renderers/webgl/WebGLProgramParameters.d.ts

```diff
@@ -0,0 +1,199 @@
+import {
+    ColorSpace,
+    Combine,
+    DepthPackingStrategies,
+    GLSLVersion,
+    ShadowMapType,
+    ToneMapping,
+} from '../../constants.js';
+
+export interface WebGLProgramParameters {
+    isWebGL2: boolean;
+
+    shaderID: string;
+    shaderType: string;
+    shaderName: string;
+
+    vertexShader: string;
+    fragmentShader: string;
+    defines?: { [define: string]: string | number | boolean };
```

Comment:
nit: this will always exist, but it may be `undefined`
```suggestion
    defines: { [define: string]: string | number | boolean } | undefined;
```